// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.mms.ecu";
option java_outer_classname = "EcuProto";
option objc_class_prefix = "ECU";

package ecu_grpc;

// The greeting service definition.
service EcuService {
  rpc reset (Empty) returns (State) {}
  rpc offline (Empty) returns (State) {}
  rpc getState (Empty) returns (State) {}
  rpc setState (State) returns (State) {}
  rpc ping (Empty) returns (State) {}
  rpc sendMsg (InterprocMsg) returns (State) {}
}

message Empty {
}

message State {
}

message InterprocMsg {

  enum InterprocCommand {
    UNKNOWN_CMD = 0;
    ECHO_CMD = 1;
    PING_CMD = 2;
    ACK_CMD = 3;
    RESET_CMD = 4;
    STATUS_CMD = 5;
    SYNC_CMD = 6;
    FIRMWARE_UPDATE_CMD = 7;
    PROGRAM_UPDATE_CMD = 8;
    VALUE_CMD = 9;
    PARAM_GET_CMD = 10;
    PARAM_SET_CMD = 11;
  }

  InterprocCommand command = 1;
  uint64 data = 2;
  uint32 checksum = 3;
}
